object_model/artifact.py

```diff
@@ -365,57 +365,179 @@ def pull(self, manifest={}):
                 else:
-                    out = []
-                    with open(dst + '/' + self.loc, 'r') as f:
-                        for i in range(head):
-                            out.append(f.readline())
-                    out = func(out)
-
-                os.chdir(original_dir)
-                # [os.remove(i) for i in self.xp_state.ghostFiles] # We want to keep resources intact
-            except Exception as e:
-                print(e)
-                out = e
-        # TODO: add case for peek next.
-
-        # Delete the temporary versioning directory created.
-        try:
-            rmtree(self.xp_state.versioningDirectory)
-        except:
-            pass
-        # Reset versioning directory to original flor.d directory.
-        self.xp_state.versioningDirectory = trueVersioningDir
+                    literal.v = old_binding
+            self.i = 0
+
+            # Unpickle and read the file to output.
+            if util.isPickle(dst + "/" + self.loc):
+                out = func(util.unpickle(dst + '/' + self.loc))
+            else:
+                out = []
+                with open(dst + '/' + self.loc, 'r') as f:
+                    for i in range(head):
+                        out.append(f.readline())
+                out = func(out)
+
+        except Exception as e:
+            try:
+                intermediateFiles = set(self.loclist) - userDefFiles
+                for file in intermediateFiles:
+                    if os.path.exists(file):
+                        os.remove(file)
+            except Exception as ee:
+                print(ee)
+            self.xp_state.literals = []
+            self.xp_state.ghostFiles = set([])
+            raise e
+
+        intermediateFiles = set(self.loclist) - userDefFiles
+        for file in intermediateFiles:
+            os.remove(file)
+        os.remove('.' + experimentName + '.flor')
+        original_dir = os.getcwd()
+
+        # Create versioning directory flor.d
+        if not os.path.isdir(self.xp_state.versioningDirectory):
+            os.mkdir(self.xp_state.versioningDirectory)
+
+        # Copy all relevant files into the versioning directory.
+        # for file in (userDefFiles & (set(self.loclist) | set(self.scriptNames))):
+        #     copyfile(file, self.xp_state.versioningDirectory + '/' + file)
+
+        os.chdir(self.xp_state.versioningDirectory)
+
+        self.xp_state.literals = []
+        self.xp_state.ghostFiles = set([])
+
+        moveBackFlag = False
+
+        # Move back  .git file in the versioning directory to the /tmp folder.
+        # Delete experiment folder in versioning directory.
+        if os.path.exists(self.xp_state.versioningDirectory + '/' + experimentName):
+            move(self.xp_state.versioningDirectory + '/' + experimentName + '/.git', '/tmp/')
+            rmtree(self.xp_state.versioningDirectory + '/' + experimentName)
+            moveBackFlag = True
+
+        if manifest:
```

Comment:
Not sure we need manifest in peek, this is for pull.

Reply:
Okay I removed it!